Source/monster.h

```diff
@@ -260,6 +260,13 @@ struct Monster { // note: missing field _mAFNum
 	 */
 	void Petrify();
 
+	/**
+	 * @brief Returns the network identifier for this monster
+	 *
+	 * This is currently the index into the Monsters array, but may change in the future.
+	 */
+	size_t GetId();

```

Comment:
`readability-identifier-naming.MethodCase: camelBack` The existing methods use the wrong case style and haven't been fixed yet, new methods should use the new style. Also probably want to address modernize-use-nodiscard. Finally, this method doesn't change state so should be declared const.
```suggestion
	[[nodiscard]] size_t getId() const;
```

Reply:
Hi, the compiler throws this error when I add `const`:

```
error: no matching function for call to 'distance'
        return std::distance(&Monsters[0], this);
```

It could be related to the iterator being mutable

Reply:
`return std::distance(const_cast<const Monster *>(&Monsters[0]), this);`

Reply:
https://en.cppreference.com/w/cpp/utility/as_const could also be useful if it's supported on GCC6.5? haven't tested that

Reply:
From the changelog as_const appears to have been added in 7.0

Reply:
`std::distance<const Monster *>(&Monsters[0], this);` should also work

Reply:
yep, already doing it for players

Reply:
PR updated